File: src/main/java/gg/mineral/practice/kit/KitCreator.java
package gg.mineral.practice.kit;

import org.bukkit.GameMode;

import gg.mineral.practice.entity.PlayerStatus;
import gg.mineral.practice.entity.Profile;
import gg.mineral.practice.inventory.SubmitAction;
import gg.mineral.practice.managers.KitEditorManager;
import gg.mineral.practice.scoreboard.impl.KitCreatorScoreboard;
import gg.mineral.practice.util.PlayerUtil;
import gg.mineral.practice.util.messages.impl.ChatMessages;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.val;

@AllArgsConstructor
public class KitCreator {
    Profile profile;
    @Getter
    SubmitAction submitAction;

    public void save() {
        profile.getDuelSettings()
                .setKit(new Kit(profile.getInventory().getContents(), profile.getInventory().getArmorContents()));
        profile.getPlayer().closeInventory();
        ChatMessages.KIT_SAVED.send(profile.getPlayer());
    }

    public void start() {
        profile.setPlayerStatus(PlayerStatus.KIT_CREATOR);
        profile.setScoreboard(KitCreatorScoreboard.INSTANCE);
        PlayerUtil.teleportNoGlitch(profile, KitEditorManager.getLocation());
        profile.getInventory().setInventoryClickCancelled(false);
        profile.getInventory().clear();

        for (val player : profile.getPlayer().getWorld().getPlayers())
            profile.removeFromView(player.getUniqueId());

        profile.setGameMode(GameMode.CREATIVE);
    }
}

File: src/main/kotlin/gg/mineral/practice/kit/KitCreator.kt
package gg.mineral.practice.kit

import gg.mineral.practice.entity.PlayerStatus
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.inventory.SubmitAction
import gg.mineral.practice.managers.KitEditorManager.bukkitLocation
import gg.mineral.practice.scoreboard.impl.KitCreatorScoreboard
import gg.mineral.practice.util.PlayerUtil
import gg.mineral.practice.util.messages.impl.ChatMessages

class KitCreator(val profile: Profile, val submitAction: SubmitAction) {
    fun save() {
        profile.duelSettings
            .kit = Kit(profile.inventory.contents, profile.inventory.armorContents)
        profile.player.closeInventory()
        ChatMessages.KIT_SAVED.send(profile.player)
    }

    fun start() {
        profile.playerStatus = PlayerStatus.KIT_CREATOR
        profile.scoreboard = KitCreatorScoreboard.INSTANCE
        PlayerUtil.teleportNoGlitch(profile, bukkitLocation)
        profile.inventory.inventoryClickCancelled = false
        profile.inventory.clear()
    }
}
